client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -226,6 +226,12 @@ function mediaButton( editor ) {
 			if ( current.media.transient ) {
 				transients++;
 				isTransientDetected = true;
+
+				// Mark the image as a transient in upload
+				editor.dom.$( img ).toggleClass( 'is-transient', true );
```

Comment:
Not now. 
It's just preparing the code so we can test transitions live in a separate CSS-only PR.